xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Document.java

```diff
@@ -1157,6 +1158,24 @@ public int createNewObject(String classname) throws XWikiException
         return index;
     }
 
+    /**
+     * Creates a new XObject from the given class reference.
+     *
+     * @param classReference the reference to the class of the XObject to be created
+     * @return the index of the newly created XObject
+     * @throws XWikiException if an error occurs while creating the XObject
+     * @since 17.4.0RC1
+     */
+    @Unstable
+    public int createNewObject(EntityReference classReference) throws XWikiException
```

Comment:
Is it really worth introducing this method ? I don't find it very useful (compared to simply always use #newObject).